Domain/SecurityIdentityRetrievalStrategy.php

```diff
@@ -39,37 +39,54 @@ public function __construct(RoleHierarchyInterface $roleHierarchy, Authenticatio
 
     /**
      * {@inheritdoc}
+     *
+     * @return RoleSecurityIdentity[]
      */
-    public function getSecurityIdentities(TokenInterface $token)
+    public function getSecurityIdentities(?TokenInterface $token)
     {
         $sids = [];
 
         // add user security identity
-        if (!$token instanceof AnonymousToken) {
+        if (!$token instanceof AnonymousToken && null !== $token) {
             try {
                 $sids[] = UserSecurityIdentity::fromToken($token);
             } catch (\InvalidArgumentException $e) {
                 // ignore, user has no user security identity
             }
         }
 
-        // add all reachable roles
-        foreach ($this->roleHierarchy->getReachableRoleNames($token->getRoleNames()) as $role) {
-            $sids[] = new RoleSecurityIdentity($role);
+        if (null !== $token) {
+            // add all reachable roles
+            foreach ($this->roleHierarchy->getReachableRoleNames($token->getRoleNames()) as $role) {
+                $sids[] = new RoleSecurityIdentity($role);
+            }
         }
 
+        $anonymousRole = \defined('\Symfony\Component\Security\Core\Authorization\Voter\AuthenticatedVoter::PUBLIC_ACCESS') ? AuthenticatedVoter::PUBLIC_ACCESS : AuthenticatedVoter::IS_AUTHENTICATED_ANONYMOUSLY;
         // add built-in special roles
         if ($this->authenticationTrustResolver->isFullFledged($token)) {
             $sids[] = new RoleSecurityIdentity(AuthenticatedVoter::IS_AUTHENTICATED_FULLY);
             $sids[] = new RoleSecurityIdentity(AuthenticatedVoter::IS_AUTHENTICATED_REMEMBERED);
-            $sids[] = new RoleSecurityIdentity(AuthenticatedVoter::IS_AUTHENTICATED_ANONYMOUSLY);
+            $sids[] = new RoleSecurityIdentity($anonymousRole);
         } elseif ($this->authenticationTrustResolver->isRememberMe($token)) {
             $sids[] = new RoleSecurityIdentity(AuthenticatedVoter::IS_AUTHENTICATED_REMEMBERED);
-            $sids[] = new RoleSecurityIdentity(AuthenticatedVoter::IS_AUTHENTICATED_ANONYMOUSLY);
-        } elseif ($this->authenticationTrustResolver->isAnonymous($token)) {
-            $sids[] = new RoleSecurityIdentity(AuthenticatedVoter::IS_AUTHENTICATED_ANONYMOUSLY);
+            $sids[] = new RoleSecurityIdentity($anonymousRole);
+        } elseif ($this->isNotAuthenticated($token)) {
+            $sids[] = new RoleSecurityIdentity($anonymousRole);
         }
 
         return $sids;
     }
+
+    /**
+     * @psalm-suppress UndefinedInterfaceMethod
+     */
```

Comment:
```suggestion
```

We don't have to suppress errors. In the Symfony Psalm set-up, psalm is run first on the target branch, and then on the PR branch. So any errors present on the main branch isn't reported in subsequent PRs. So we'll ignore the error now and it'll be fine after.

Reply:
fixed!